### PR TITLE
Add defect to testresult

### PR DESCRIPTION
**Pull request by cuongld2**
Dear allan,
I would like to make PR for adding defect to the test result with decorator
Probably resolve this issue : https://github.com/allankp/pytest-testrail/issues/45
Thanks

### CODE REVIEW

File: pytest_testrail/conftest.py [hunk 1]
```diff
@@ -123,8 +123,10 @@ def pytest_configure(config):
                 assign_user_id=config_manager.getoption('tr-testrun-assignedto-id', 'assignedto_id', 'TESTRUN'),
                 project_id=config_manager.getoption('tr-testrun-project-id', 'project_id', 'TESTRUN'),
                 suite_id=config_manager.getoption('tr-testrun-suite-id', 'suite_id', 'TESTRUN'),
-                include_all=config_manager.getoption('tr-testrun-suite-include-all', 'include_all', 'TESTRUN', is_bool=True, default=False),
-                cert_check=config_manager.getoption('tr-no-ssl-cert-check', 'no_ssl_cert_check', 'API', is_bool=True, default=True),
+                include_all=config_manager.getoption('tr-testrun-suite-include-all', 'include_all', 'TESTRUN',
+                                                     is_bool=True, default=False),
+                cert_check=config_manager.getoption('tr-no-ssl-cert-check', 'no_ssl_cert_check', 'API', is_bool=True,
+                                                    default=True),
```
Comment (PeterNgTr) [inline pytest_testrail/conftest.py]: not sure if this is by purpose to break them into new line. :)

File: pytest_testrail/plugin.py [hunk 2]
```diff
@@ -100,6 +112,16 @@ def clean_test_ids(test_ids):
     return [int(re.search('(?P<test_id>[0-9]+$)', test_id).groupdict().get('test_id')) for test_id in test_ids]
 
 
+def clean_test_defects(defect_ids):
+    """
+        Clean pytest marker containing testrail testcase ids.
```
Comment (PeterNgTr) [inline pytest_testrail/plugin.py]: I guess this should be `Clean pytest marker containing defect ids.`

File: tests/test_plugin.py [hunk 3]
```diff
@@ -151,51 +154,54 @@ def get_result(self):
...
+            'duration': 2,
+            'defects': None
         },
         {
             'case_id': 5678,
             'status_id': TESTRAIL_TEST_STATUS["failed"],
             'comment': "An error",
-            'duration': 2
+            'duration': 2,
+            'defects': None
```
Comment (PeterNgTr) [inline tests/test_plugin.py]: maybe you can add tests that have defects?

File: pytest_testrail/plugin.py [hunk 4]
```diff
@@ -185,16 +207,28 @@ def pytest_runtest_makereport(self, item, call):
...
-
             if rep.when == 'call' and testcaseids:
-                self.add_result(
-                    clean_test_ids(testcaseids),
-                    get_test_outcome(outcome.get_result().outcome),
-                    comment=rep.longrepr,
-                    duration=rep.duration
-                )
+                if defectids != None:
+                    print('HAAAAAAAAAAAA Defects are :', str(clean_test_defects(defectids)).replace('[', '').replace(']', '').replace("'", ''))
```
Comment (PeterNgTr) [inline pytest_testrail/plugin.py]: 😄

File: tests/test_plugin.py (hunk 3 above)
Comment (cuongld2) [reply tests/test_plugin.py]: Yeah, you're right.
I've just added some meaningful defects value.
Thanks Peter

File: pytest_testrail/plugin.py (hunk 2 above)
Comment (cuongld2) [reply pytest_testrail/plugin.py]: Yeah done that. :d

File: pytest_testrail/plugin.py (hunk 4 above)
Comment (cuongld2) [reply pytest_testrail/plugin.py]: just remove it :dagger:

File: setup.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ def read_file(fname):
     install_requires=[
         'pytest>=3.6',
         'requests>=2.20.0',
-        'simplejson',
+        'simplejson', 'freezegun', 'mock'
```
Comment (PeterNgTr) [inline setup.py]: just remove unused libs :)

File: tests/test_plugin.py [hunk 6]
```diff
@@ -3,12 +3,13 @@
...
 import pytest
-
+import sys
+import os
 from pytest_testrail import plugin
 from pytest_testrail.plugin import PyTestRailPlugin, TESTRAIL_TEST_STATUS
 from pytest_testrail.testrail_api import APIClient
 
-
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
```
Comment (PeterNgTr) [inline tests/test_plugin.py]: could you enlighten me what this line does?

Comment (cuongld2) [reply tests/test_plugin.py]: yeah that's unnecessary code.
Thanks Peter

File: setup.py (hunk 5 above)
Comment (cuongld2) [reply setup.py]: removed

File: tests/test_plugin.py [hunk 7]
```diff
@@ -3,7 +3,8 @@
 from freezegun import freeze_time
 from mock import call, create_autospec
 import pytest
-
+import sys
+import os
```
[the remark this answers is not in the document]
Comment (cuongld2) [reply tests/test_plugin.py]: Yeah. These imports are unnecessary.
I've just removed it.

File: pytest_testrail/plugin.py [hunk 8]
```diff
@@ -221,22 +254,24 @@ def pytest_sessionfinish(self, session, exitstatus):
 
     # plugin
 
-    def add_result(self, test_ids, status, comment='', duration=0):
+    def add_result(self, test_ids, status, comment='', duration=0, defects=None):
```
Comment (allankp) [inline pytest_testrail/plugin.py]: `defects` should be before `duration`

Comment (cuongld2) [reply pytest_testrail/plugin.py]: Dear allankp,
Thanks for the comment.
I've just changed the order.

**Comment by cuongld2** [comment]
> 
> 
> Can you please update the README with the new flag please

I've just updated the README with example for using pytestrail.defects.
Please have a look.
Thank you!

**Pull request merged**